sensors/simsensor_common/simsensor_generic.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Generic functions for all sensor simulators
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "simsensor_generic.h"
+
+#include "event_queue.h"
+#include "simsensor_reader.h"
+
+#include <stdlib.h>
+#include <errno.h>
+#include <time.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include <string.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+
+typedef struct {
+	char name[SIMSENS_NAME_MAX_LEN + 1];
```

Comment:
There is ambiguity between `simsens_ctx_t:name` and `simsens_settings_t:name` when it comes to handling memory. I think it is possible to solely rely on `const char*` as `name` field type in both cases. It is always passed as string literal and there is no way to pass another name to simsensor in runtime.

This will also make length checking not necessary in alloc.